src/main/java/ci/PersistentLogs.java

```
@@ -0,0 +1,117 @@
+package ci;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Date;
+import java.util.Arrays;
+import java.util.ArrayList;
+
+import org.json.JSONTokener;
+import org.json.JSONObject;
+
+
+public class PersistentLogs {
+    public static String path;  // The relative path to the folder where the logs are stored.
+    public int build_number;    // The build number for easier ordering of logs and unique identifiers
+
+    /**
+     * Constructor for the PersistentLogs class.
+     * @param path_to_logs_folder - relative path to a place where the logs should be stored.
+     */
+    public PersistentLogs(String path_to_logs_folder) {
+        this.path = path_to_logs_folder;
+        File[] files = all_logs();
+        if (files.length == 0) {
+            this.build_number = 0;
+        } else {
+            // By grabbing the latest log, the counting is uninterrupted in the case of accidental log deletions.
+            this.build_number = Integer.parseInt(files[files.length - 1].getName().split("_")[0]) + 1;
+        }
+    }
+
+    /**
+     * Add a log file with JSON data taken from the parameter log_entry.
+     * @param log_entry - The Log_entry object containing data to be written to the log file
+     */
+    public void add_log(Log_entry log_entry) {
+        try {
+            File log_file = new File(path + "/" + this.build_number + "_" + log_entry.generate_log_file_name());
+            this.build_number++;
+            if (!log_file.exists()) {
+                log_file.createNewFile();
+            }
+            FileWriter fw = new FileWriter(log_file);
+            fw.write(log_entry.toString());
+            fw.close();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```

Instead of catching exceptions here and silently failing, let the caller handle any errors so they can act accordingly.

---

Good catch, this was probably done to satisfy the linter. Will do the proper "throws" rework